.github/workflows/schedule-dependencies.yml

```diff
@@ -30,9 +30,35 @@ jobs:
     - name: Install dependencies
       run: |
         uv pip install wheel --system
-        uv pip install ${{ matrix.pre-release-dependencies }} scikit-lego --system
+        uv pip install ${{ matrix.pre-release-dependencies }} --system -e ".[test-dep]"
         uv pip freeze
     - name: Test with pytest
+      run: pytest -n auto --disable-warnings --cov=sklego -m "not cvxpy and not formulaic and not umap"
+
+  cron-extra:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: ["3.10"]
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        pre-release-dependencies: ["--pre", ""]
+    steps:
+    - name: Checkout source code
+      uses: actions/checkout@v4
+    - name: Install uv (Unix)
+      if: runner.os != 'Windows'
+      run: curl -LsSf https://astral.sh/uv/install.sh | sh
+    - name: Install uv (Windows)
+      if: runner.os == 'Windows'
+      run: powershell -c "irm https://astral.sh/uv/install.ps1 | iex"
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
       run: |
-        uv pip install -e ".[test]" --system
-        make test
+        uv pip install wheel --system
+        uv pip install ${{ matrix.pre-release-dependencies }} -e ".[test-all]" --system
+        uv pip freeze
+    - name: Test with pytest
+      run: pytest -n auto --disable-warnings --cov=sklego -m "cvxpy or formulaic or umap"
```

Comment:
I wonder ... is the `-m` part needed here? Don't we test for everything by default?

Reply:
Oh true, by default everything is tested, with such flag just the _extras_. I was thinking doing a sort of "delta".

Reply:
That can still be a good idea. I guess we could first do a run without those marked tests and then run a step for each of them. Just make sure that you always run each step.

Reply:
Split it using matrix, it gets a little spammy but very detailed ([example](https://github.com/FBruzzesi/scikit-lego/actions/runs/8757546995))